### DISCUSSION TITLE
Extracting heads from Modflow binary files using flopy

### DISCUSSION THREAD

**Discussion by zargol7419** [modflowpy/flopy#1567] [Q&A] [answered]
Hi everybody
I wanna extract heads derived from running an existing modflow model by using flopy package. I have executed different codes but I haven't received my expected answer. It seems that expected heads are not calculated before or I have executed a wrong code. I can't understand what is the problem.
I want to save all calculated heads in an array and then use them for another calculation. Could you please provide me with a solution for this problem.
Here is my executed codes:
**First one:**
![image](https://user-images.githubusercontent.com/103249900/193800214-e7b496db-17a5-471c-868f-acb5cb06bc86.png)
**Second one:**
![image](https://user-images.githubusercontent.com/103249900/193800333-c3c430b8-1a46-4f4d-8171-a662a63fc520.png)
 Thanks in advance for your answers.

**Comment by jdhughes-dev** [comment, marked as answer]
Your first example has extracted the heads for all times. The data has dimensions `[output_times, nlay, nrow, ncol]`, where `output_times` is the number of times you have output binary head data (specified in the oc file). It looks like you are using MODFLOW-2000, so the `-999` values are either the `HNOFLO` or `HDRY` value you have specified in your input files. You can retrieve the head for individual times using [`.get_data()`](https://flopy.readthedocs.io/en/3.3.5/source/flopy.utils.binaryfile.html#flopy.utils.binaryfile.CellBudgetFile.get_data).

**Reply by zargol7419** [reply]
Thank you so much. I understood what should I do. I really appreciate your cooperation.
